---
title: "FRAM OBSERVATORY -- Remote Access Samplers -- 16S rRNA amplicons (run JB7R9)"
---

This markdown describes the processing of 16S rRNA amplicon sequences, originating from year-round sampling by "Remote Access Samplers" within the FRAM / HAUSGARTEN LTER. Raw fastq files are available under ENA umbrella PRJEB43905.

Primer clipping using *Cutadapt* 

```{console}

######################################
## FASTQ DOWNLOAD ##
######################################

# Set working directory
cd /isibhv/projects/FRAMdata/MolObs/WaterCol_RAS_ampTimeseries/bacteria/
  
# Fetch fastqs from TapeArchive via custom script
# Names listed in txtfile
/isibhv/projects/p_bioinf2/miseq/copy-files-from-tape.sh files_JB7R9.txt

# Rename / remove MiSeq run-ID 
rename 'M00493_512_000000000-JB7R9' 'JB7R9' *
cd JB7R9
mkdir Original 
mv *gz Original

######################################
## PRIMER CLIPPING ##
######################################

module load /albedo/soft/modules/bio/cutadapt/4.4

bash ./../../../software/cutadapt.sh ./Original GTGYCAGCMGCCGCGGTAA CCGYCAATTYMTTTRAGTTT

# test rename
cd Clipped
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/_PS121_RAS_DNA_[0-9]{2,3}_bactV4V5_S[0-9]{1,2}_L001/,"_clip");print}'`; echo -e $i $nname; done
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/_PS114_RAS_DNA_[0-9]{2,3}_bactV4V5_S[0-9]{1,2}_L001/,"_clip");print}'`; echo -e $i $nname; done

# if looking OK - execute:  
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/_PS121_RAS_DNA_[0-9]{2,3}_bactV4V5_S[0-9]{1,2}_L001/,"_clip");print}'`; mv $i $nname; done
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/_PS114_RAS_DNA_[0-9]{2,3}_bactV4V5_S[0-9]{1,2}_L001/,"_clip");print}'`; mv $i $nname; done

# write sampleNames for dada
ls -1 *R1_001.fastq.gz | sed 's/_R1_001\.fastq.gz//' > ../sampleNames.txt

```

*Now we go into DADA mode!*

# done in RStudio within AWI-VM
# provided IP address opened in browser
# adjust for your own system 

```{r, eval=F}

require(dada2)
require(ShortRead)
require(ggplot2)
require(gridExtra)

##########################################

# setwd 
setwd("/isibhv/projects/FRAMdata/MolObs/WaterCol_RAS_ampTimeseries/bacteria/JB7R9")

# list files
path <- "/isibhv/projects/FRAMdata/MolObs/WaterCol_RAS_ampTimeseries/bacteria/JB7R9/Clipped"
fns <- list.files(path)
fns

# ensure fwd/rev reads  in same order
fnFs <- sort(list.files(path, pattern="_R1_001.fastq.gz"))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq.gz"))

# Define sample names
sampleNames <- sort(read.table(
  "sampleNames.txt", 
  h=F, stringsAsFactors=F)$V1)

# Specify the full path to the fnFs and fnRs
fnFs <- file.path(path, fnFs)
fnRs <- file.path(path, fnRs)

#################################

# Quality check
QualityProfileFs <- list()
for(i in 1:length(fnFs)) {QualityProfileFs[[i]] <- list()
  QualityProfileFs[[i]][[1]] <- plotQualityProfile(fnFs[i])}
pdf("QualityProfileForward.pdf")
for(i in 1:length(fnFs)) {do.call("grid.arrange", 
    QualityProfileFs[[i]])}
dev.off()
rm(QualityProfileFs)

QualityProfileRs <- list()
for(i in 1:length(fnRs)) {
  QualityProfileRs[[i]] <- list()
  QualityProfileRs[[i]][[1]] <- plotQualityProfile(
    fnRs[i])}
pdf("QualityProfileReverse.pdf")
for(i in 1:length(fnRs)) {do.call("grid.arrange", 
  QualityProfileRs[[i]])}
dev.off()
rm(QualityProfileRs)
# rev-reads of medium/strange quality
# several with quality drop around 50nt
# these are removed after filtering, so OK

# Prepare for fastq filtering
filt_path <- file.path(path, "../Filtered")
if(!file_test("-d", filt_path)) dir.create(filt_path)
filtFs <- file.path(
  filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(
  filt_path, paste0(sampleNames, "_R_filt.fastq"))

#################################

# Filtering
out <- filterAndTrim(
  fnFs, 
  filtFs, 
  fnRs, 
  filtRs,
  truncLen = c(230, 195),
  maxN = 0,
  minQ = 2,
  maxEE = c(3, 3), 
  truncQ = 0, 
  rm.phix = T,
  compress = F,
  multithread = 12)

# Should retain >70% - 0.68 here -- barely OK...
head(out)
summary(out[, 2]/out[, 1])

#################################

# Quality check 
QualityProfileFs.filt <- list()
for(i in 1:length(filtFs)) {
  QualityProfileFs.filt[[i]] <- list()
  QualityProfileFs.filt[[i]][[1]] <- plotQualityProfile(
    filtFs[i])}
pdf("QualityProfileForwardFiltered.pdf")
for(i in 1:length(filtFs)) {do.call("grid.arrange", 
    QualityProfileFs.filt[[i]])}
dev.off()
rm(QualityProfileFs.filt)

QualityProfileRs.filt <- list()
for(i in 1:length(filtRs)) {
  QualityProfileRs.filt[[i]] <- list()
  QualityProfileRs.filt[[i]][[1]] <- plotQualityProfile(
    filtRs[i])}
pdf("QualityProfileReverseFiltered.pdf")
for(i in 1:length(filtRs)) {  do.call("grid.arrange", 
    QualityProfileRs.filt[[i]])}
dev.off()
rm(QualityProfileRs.filt)

#################################

# Learn errors 
errF <- learnErrors(
  filtFs, multithread=12, 
  randomize=T, verbose=1, MAX_CONSIST=20)
errR <- learnErrors(
  filtRs, multithread=12, 
  randomize=T, verbose=1, MAX_CONSIST=20)

# Plot error profiles
pdf("ErrorProfiles.pdf")
plotErrors(errF, nominalQ = T)
plotErrors(errR, nominalQ = T)
dev.off()
# convergence after 6-6 rounds - ok!
# few outliers outside black line - ok!

# Dereplication 
derepFs <- derepFastq(filtFs, verbose=T)
derepRs <- derepFastq(filtRs, verbose=T)

# Rename by clip-filenames
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

# Denoising
dadaFs <- dada(
  derepFs, err=errF, multithread=24, pool=T)
dadaRs <- dada(
  derepRs, err=errR, multithread=24, pool=T)

#################################

# Read merging
mergers <- mergePairs(
  dadaFs, 
  derepFs, 
  dadaRs,
  derepRs,
  minOverlap=10,
  verbose=T,
  propagateCol = c(
    "birth_fold", 
    "birth_ham"))

# Create sequence table
seqtab <- makeSequenceTable(mergers)

# 9106 sequences
dim(seqtab) 

# Save
saveRDS(seqtab, "seqtab_bac_JB7R9.rds")

# Summary stats
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(
  dadaFs, getN), sapply(mergers, getN), 
  rowSums(seqtab))
colnames(track) <- c(
  "input","filtered","denoised",
  "merged","tabled")
rownames(track) <- sampleNames
track <- data.frame(track)
head(track)

write.table(track, 
  "dadastats_bac_JB7R9.txt", 
  quote=F, sep="\t")

#################################

save.image("RAS_bac_JB7R9.Rdata")

```

Sequence tables from all RAS MiSeq runs are combined, chimera-checked and taxonomy-assigned using the MergeChimTax Rscript.