---
title: "FRAM OBSERVATORY -- Remote Access Samplers -- Processing of 16S rRNA amplicons (run JR82J)"
---

This markdown describes the processing of 16S rRNA amplicon sequences, originating from continuous year-round sampling by "Remote Access Samplers" in the framework of the FRAM LTER  (https://www.awi.de/en/expedition/observatories/ocean-fram.html). The corresponding raw fastq files are available under ENA accession XXX. 

First we remove primers using *Cutadapt*  

```{console}

######################################
## FILE PREPARATION ##
######################################

# Fastq files listed in "files2cp.txt" are copied from AWI tape archive using in-house script  

cd /isibhv/projects/FRAMdata/FRAM_MicrObs/WaterCol_RAS_1620/bacteria/JR82J/
mkdir Original
cd Original

/global/AWIsoft/bio/scripts/getFastqsFromTape.sh ../files2cp.txt
/global/AWIsoft/bio/scripts/renameFastqs.sh
cd ..

######################################
## PRIMER CLIPPING ##
######################################

module load bio/cutadapt/3.2
bash ./../../../software/cutadapt.sh ./Original GTGYCAGCMGCCGCGGTAA CCGYCAATTYMTTTRAGTTT

# test rename
cd Clipped
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/S[0-9]{1,3}_L001/,"clip");print}'`; echo -e $i $nname; done
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/_21_4_DNA/,"");print}'`; echo -e $i $nname; done

# if looking OK - execute:  
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/S[0-9]{1,3}_L001/,"clip");print}'`; mv $i $nname; done
for i in *fastq.gz; do nname=`echo $i | awk '{gsub(/_21_4_DNA/,"");print}'`; mv $i $nname; done

# write sampleNames for dada
ls -1 *R1_001.fastq.gz | sed 's/_R1_001\.fastq.gz//' > ../sampleNames.txt

```

*Now we go into DADA mode!*

# done in RStudio within AWI-VM
# provided IP address opened in browser
# adjust for your own system 

```{r, eval=F}

require(dada2)
require(ShortRead)
require(ggplot2)
require(gridExtra)

##########################################

# setwd 
setwd("/isibhv/projects/FRAMdata/FRAM_MicrObs/WaterCol_RAS_1620/bacteria/JR82J")

# list files
path <- "/isibhv/projects/FRAMdata/FRAM_MicrObs/WaterCol_RAS_1620/bacteria/JR82J/Clipped"
fns <- list.files(path)
fns

# ensure fwd/rev reads  in same order
fnFs <- sort(list.files(path, pattern="_R1_001.fastq.gz"))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq.gz"))

# Define sample names
sampleNames <- sort(read.table(
  "sampleNames.txt", 
  h=F, stringsAsFactors=F)$V1)

# Specify the full path to the fnFs and fnRs
fnFs <- file.path(path, fnFs)
fnRs <- file.path(path, fnRs)

#################################

# Quality check
QualityProfileFs <- list()
for(i in 1:length(fnFs)) {QualityProfileFs[[i]] <- list()
  QualityProfileFs[[i]][[1]] <- plotQualityProfile(fnFs[i])}
pdf("QualityProfileForward.pdf")
for(i in 1:length(fnFs)) {do.call("grid.arrange", 
    QualityProfileFs[[i]])}
dev.off()
rm(QualityProfileFs)

QualityProfileRs <- list()
for(i in 1:length(fnRs)) {
  QualityProfileRs[[i]] <- list()
  QualityProfileRs[[i]][[1]] <- plotQualityProfile(
    fnRs[i])}
pdf("QualityProfileReverse.pdf")
for(i in 1:length(fnRs)) {do.call("grid.arrange", 
  QualityProfileRs[[i]])}
dev.off()
rm(QualityProfileRs)
# many rev-reads of very bad quality!!

# Prepare for fastq filtering
filt_path <- file.path(path, "../Filtered")
if(!file_test("-d", filt_path)) dir.create(filt_path)
filtFs <- file.path(
  filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(
  filt_path, paste0(sampleNames, "_R_filt.fastq"))

#################################

# Filter depending on expected overlap
# truncLen lowered based on low quality esp. of rev-reads
out <- filterAndTrim(
  fnFs, 
  filtFs, 
  fnRs, 
  filtRs,
  truncLen = c(230, 195),
  maxN = 0,
  minQ = 2,
  maxEE = c(3, 3), 
  truncQ = 0, 
  rm.phix = T,
  compress = F,
  multithread = 24)
head(out)
summary(out[, 2]/out[, 1])
# should be retaining >70% (0.7) still OK...

#################################

# Quality check 
QualityProfileFs.filt <- list()
for(i in 1:length(filtFs)) {
  QualityProfileFs.filt[[i]] <- list()
  QualityProfileFs.filt[[i]][[1]] <- plotQualityProfile(
    filtFs[i])}
pdf("QualityProfileForwardFiltered.pdf")
for(i in 1:length(filtFs)) {do.call("grid.arrange", 
    QualityProfileFs.filt[[i]])}
dev.off()
rm(QualityProfileFs.filt)

QualityProfileRs.filt <- list()
for(i in 1:length(filtRs)) {
  QualityProfileRs.filt[[i]] <- list()
  QualityProfileRs.filt[[i]][[1]] <- plotQualityProfile(
    filtRs[i])}
pdf("QualityProfileReverseFiltered.pdf")
for(i in 1:length(filtRs)) {  do.call("grid.arrange", 
    QualityProfileRs.filt[[i]])}
dev.off()
rm(QualityProfileRs.filt)

#################################

# Learn errors 
errF <- learnErrors(
  filtFs, multithread=24, 
  randomize=T, verbose=1, MAX_CONSIST=20)
errR <- learnErrors(
  filtRs, multithread=24, 
  randomize=T, verbose=1, MAX_CONSIST=20)

# Plot error profiles
pdf("ErrorProfiles.pdf")
plotErrors(errF, nominalQ = T)
plotErrors(errR, nominalQ = T)
dev.off()
# convergence after 6-6 rounds - ok!
# few outliers outside black line - ok!

# Dereplication 
derepFs <- derepFastq(filtFs, verbose=T)
derepRs <- derepFastq(filtRs, verbose=T)

# Rename by clip-filenames
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

# Denoising
dadaFs <- dada(
  derepFs, err=errF, multithread=24, pool=T)
dadaRs <- dada(
  derepRs, err=errR, multithread=24, pool=T)

#################################

# Read merging
mergers <- mergePairs(
  dadaFs, 
  derepFs, 
  dadaRs,
  derepRs,
  minOverlap=10,
  verbose=T,
  propagateCol = c(
    "birth_fold", 
    "birth_ham"))

# create sequence table
seqtab <- makeSequenceTable(mergers)
dim(seqtab) # identified 4157 sequences
saveRDS(seqtab, 
  "seqtab_bac_JR82J.rds")

# save another seqtab for merging/chim/tax
saveRDS(seqtab, 
  "../seqtab_bac_JR82J.rds")

# summary stats
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(
  dadaFs, getN), sapply(mergers, getN), 
  rowSums(seqtab))
colnames(track) <- c(
  "input","filtered","denoised",
  "merged","tabled")
rownames(track) <- sampleNames
track <- data.frame(track)
head(track)

write.table(track, 
  "dadastats_bac_JR82J.txt", 
  quote=F, sep="\t")

#################################

save.image("RAS_bac_JR82J.Rdata")

```

Sequence tables from all RAS amplicons are combined, chimera-checked and assigned taxonomy using the MergeChimTax Rscript.